template_agent/src/api.py

```diff
@@ -78,9 +82,9 @@ async def receive():
 
                 request = Request(request.scope, receive)
             except Exception as e:
-                logger.warning("Failed to read request body", error=str(e))
+                logger.warning(f"Failed to read request body: {str(e)}")
 
-        logger.info("incoming_request", **request_data)
+        logger.info(f"Incoming request: {request_data}")
```

Comment:
I believe that best practice is to avoid using f-strings in logger calls:
```suggestion

        logger.info("Incoming request: %s", request_data)
```

This defers the string interpolation to the point where we know that we are going to produce a log message.  (That is, if the current log level is set to `WARNING` then this call will not produce a message in the log, because this message is an `INFO`-level message; using an f-string requires the interpreter to produce the message string _before_ the `info()` call is made, whereas using a parameterized string allows us to avoid the string processing when it's not required.)